File: scope104/src/main/java/com/petro/scope104/presentation/list/MyFragmentAdapter.java
package com.petro.scope104.presentation.list;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class MyFragmentAdapter extends FragmentPagerAdapter {
    public MyFragmentAdapter(FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int i) {
        return WorkerListFragment.newInstance(ListType.values()[i]);
    }

    @Override
    public int getCount() {
        return ListType.values().length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return ListType.values()[position].name();
    }
}

File: scope104/src/main/kotlin/com/petro/scope104/presentation/list/MyFragmentAdapter.kt
package com.petro.scope104.presentation.list

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class MyFragmentAdapter(fm: FragmentManager?) : FragmentPagerAdapter(
    fm!!
) {
    override fun getItem(i: Int): Fragment {
        return WorkerListFragment.Companion.newInstance(ListType.values()[i])
    }

    override fun getCount(): Int {
        return ListType.values().size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return ListType.values()[position].name
    }
}